tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15756,15 +15891,18 @@ void TryAddShaderAttrFromTargetProfile(Sema &S, FunctionDecl *FD,
   }
 
   HLSLShaderAttr *currentShaderAttr = FD->getAttr<HLSLShaderAttr>();
-  // Don't add the attribute if it already exists as an attribute on the decl.
-  // In the special case that the target profile is compute and the
-  // entry decl already has a node shader attr, don't do anything
+  // Don't add the attribute if it already exists as an attribute on the decl,
+  // and emit an error. In the special case that the target profile is compute
+  // and the entry decl already has a node shader attr, don't do anything
   if (currentShaderAttr) {
     llvm::StringRef currentFullName = currentShaderAttr->getStage();
     if (currentFullName != fullName) {
-      S.Diag(currentShaderAttr->getLocation(),
-             diag::err_hlsl_profile_conflicts_with_shader_attribute)
-          << fullName << profile << currentFullName << EntryPointName;
+      if (!(SM->IsCS() && strstr(currentFullName.data(), "node"))) {
```

Comment:
I'm a little confused by this logic. We don't allow node entries to be used as compute entries.

Reply:
This logic was meant to be the move from CGHLSLMS.cpp, where this diagnostic is emitted:
`Diags.getCustomDiagID(DiagnosticsEngine::Error,
                                    "Node function as global entry point must "
                                    "be compiled to compute shader target");`
I believe my logic accurately detects when this case happens, and emits a diagnostic. Is it true that we don't allow a Node function as a global entry point to be compiled to a compute shader target?

Reply:
We disallowed mixing the entry points and removed the implicit input record arguments, so I believe this code path also shouldn’t be supported.

@tex3d & @pow2clk, thoughts?

Reply:
It is slightly different from the decision we made before to disallow node and compute shader attributes on the same entry point, but I think we should keep the existing behavior that disallows using an entry point for a target that doesn't match its shader attribute.

Reply:
Fixed.